### PR TITLE
Log docker-gc activity to syslog

### LINKED ISSUE #55: output/log which containers and images have been deleted
It would be nice if the script could output in some capacity what it had removed, to be able to trace back issues where an image/container was unexpectedly removed from a host.

The current version of the script keeps some output of what it has done in the "state directory" but this is overwritten each time the script is re-run.

For instance, I would like to be able to check /var/log/cron.log to see that docker-gc deleted image 123456 three hours ago, even if I have run the script 10 times more since then.

### PR DESCRIPTION
**Pull request by mattnworb**
replaces #56 (thanks @konstruktoid !) and resolves #55.

### CODE REVIEW

**Comment by mattnworb** [comment]
@gimaker @davidxia

**Comment by mattnworb** [comment]
and @rohansingh

**Comment by davidxia** [comment]
LGTM but my bash foo isn't as good as @gimaker :)

File: docker-gc [hunk 1]
```diff
@@ -139,6 +149,35 @@ function compute_exclude_container_ids() {
         | sort -u > $EXCLUDE_CONTAINER_IDS_FILE
 }
 
+function log() {
+    msg=$1
+    if [[ $LOG_TO_SYSLOG -gt 0 ]]; then
+        logger -i -t "$SYSLOG_TAG" -p "$SYSLOG_FACILITY.$SYSLOG_LEVEL" "$msg"
+    else
+        echo $msg
```
Comment (gimaker) [inline docker-gc]: double-quote `$msg`

**Comment by gimaker** [comment]
:+1: a paranoid person would disable logging by default :)

**Comment by mattnworb** [comment]
addressed both points, thanks @gimaker

**Pull request merged**